jsdom-monkeypatch.js

```diff
@@ -0,0 +1,308 @@
+      return respecUrl;
+    }
+    else if (/\.[^\/\.]+$/.test(url.path) &&
+        !url.path.endsWith('.js') &&
+        !url.path.endsWith('.json')) {
+      //console.log(`fetch not needed for ${url.href} (not a JS/JSON file)`);
+      return null;
+    }
+    else if ((url.pathname === '/webcomponents/assets/scripts/autolink.js') ||
+        (url.href.startsWith(referrer) &&
+          !(/annotate_spec/i.test(url.pathname)) &&
+          !(/expanders/i.test(url.pathname)) &&
+          !(/bug-assist/i.test(url.pathname)) &&
+          !(/dfn/i.test(url.pathname)) &&
+          !(/section-links/i.test(url.pathname)) &&
+          !(/^\/webidl\//i.test(url.pathname)))) {
+      //console.log(`fetch useful script at ${url.href}`);
+      return url.href;
+    }
+    //console.log(`fetch not needed for ${url.href}`);
+    return null;
+  }
+
+  let urlToFetch = getUrlToFetch();
+  if (!urlToFetch) {
+    return callback(null, '');
+  }
+  fetch(urlToFetch, options)
+    .then(response => response.text())
+    .then(data => {
+      if (urlToFetch !== respecUrl) {
+          return data;
+      }
+
+      ////////////////////////////////////////////////////////////
+      // REALLY UGLY CODE WARNING
+      //
+      // Tweak Respec built code so that it can run in JSDOM.
+      //
+      // NB: Some of these lines will just break if Respec build
+      // produces slightly different code, e.g. if variables do
+      // not end up with the same name!
+      ////////////////////////////////////////////////////////////
+
+      // Remove core/highlight module because JSDOM does not yet
+      // support URL.createObjectURL
+      // https://github.com/jsdom/jsdom/issues/1721
+      // Remove core/list-sorter module because JSDOM does not yet
+      // support document.createRange
+      // https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/nodes/Document.webidl#L39
+      ["core/highlight", "core/list-sorter"]
+        .forEach(module => data = data.replace(
+          new RegExp('(define\\(\\s*"profile-w3c-common"\\s*,\\s*\\[[^\\]]+),\\s*"' + module + '"'),
+          '$1'));
+
+      // JSDOM's CSS parser does not quite like uncommon "@" rules
+      // so let's pretend they are just @media rules
+      // https://github.com/jsdom/jsdom/issues/2026
+      // (NB: this replacement is just for convenience, to avoid JSDOM reporting
+      // lengthy errors (including a full dump of the CSS) to stderr
+      data = data.replace(/@keyframes \S+? {/, '@media all {');
+      data = data.replace(/@supports \(.+?\) {/, '@media all {');
+
+      // Respec drops blank lines in Markdown, but marked.js actually
+      // needs them around <pre> tags, otherwise it produces really weird
+      // HTML (with <pre> and <p> intertwined). For some reason, this does
+      // not bother regular browsers. It does bother JSDOM though.
+      data = data.replace(/r\.createTextNode\("\\n"\)/, 'r.createTextNode("\\n\\n")');
+
```

Comment:
maybe drop `r.` to make less fragile?

Reply:
Actually, there's another call to `createTextNode("\n")` somewhere else in the code which would be modified as well. I'll stick with that code for now, but will investigate upstreaming that change to ReSpec (after all, the HTML produced is wrong in regular browsers as well, it just happens not to have any bad consequence there)